airbyte-integrations/connectors/source-google-ads/source_google_ads/spec.json

```diff
@@ -67,6 +67,32 @@
         "maximum": 1095,
         "default": 14,
         "examples": [14]
+      },
+      "custom_queries": {
+        "type": "array",
+        "items": {
+          "type": ["object", "null"],
+          "properties": {
+            "query": {
+              "type": "string",
+              "title": "User defined query to build a report by wish",
+              "examples": ["SELECT segments.ad_destination_type, campaign.advertising_channel_sub_type FROM campaign"],
+              "pattern": "(?i)^select(\\s)+[a-zA-Z0-9._,(\\s)-]*(\\s+)from(\\s+)(campaign|customer|ad_group|ad_group_ad)$"
```

Comment:
what's the reason to limit which resources a user can access? there are many resources in the API that would be nice to have, not just campaign/customer/ad_group/ad_group_ad

Reply:
if the issue is schemas then we can potentially get away with: 

1. using the google defined protobuf schema files herehttps://github.com/googleapis/googleapis/tree/master/google/ads/googleads/v8/
2. converting them to json using a converter like this https://github.com/chrusty/protoc-gen-jsonschema

Alternatively, we can use the GoogleAdsFieldService to dynamically determine the type of the fields in the query. See this page: https://developers.google.com/google-ads/api/docs/query/overview

Reply:
I will take some examples from https://github.com/googleapis/googleapis/tree/master/google/ads/googleads/v8/ to local folder to be able to use them

Reply:
I think you can actually have an easier time using the GoogleAdsFieldsService